### DISCUSSION TITLE
Trigger action on.schedule only if there are changes to the branch

### DISCUSSION THREAD

**Discussion by nheath99** [community/community#27128] [Actions]
<p>I would like to run an Action once every 24 hours, only if commits have been made to the target branch in the last 24 hours (or since the last run). This is for calculating code coverage on a large repo, which takes about 20-30 mins.</p>
<p>I can find documentation for on.schedule (<a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule" rel="nofollow noopener">https://help.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule</a>) but not for checking the date/time of the latest commit.</p>

**Comment by logankilpatrick** [comment]
<p>Hey <a class="mention" href="/nheath99">@nheath99</a>, this is not a functionary that is supported out of the box. I suggest adding in an additional piece to the action that uses Python or another language to check the git contents to see if there’s a commit in the last 24 hours. </p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/nheath99">@nheath99</a> ,</p>
<p>You can use ‘schedule’ event, and in the job check the last commit time via script in the link: <a href="https://gist.github.com/jasonrudolph/1810768" rel="nofollow noopener">https://gist.github.com/jasonrudolph/1810768</a>.</p>
<p>If the last commit is in 24hours, execute your steps, otherwise skip it(use <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" rel="nofollow noopener">if</a> expression to skip the job ).</p>
<p>Thanks.</p>

**Reply by mrcjkb** [reply]
In case anyone comes across this and needs an example, [here](https://github.com/nvim-neorocks/luarocks-tag-release#version-optional) is a snippet.

**Comment by CfirTsabari** [comment]
<p>you can do something like this:</p>
<ol>
<li>first add this job:</li>
</ol>
<pre><code class="lang-auto">  check_date:
    runs-on: ubuntu-latest
    name: Check latest commit
    outputs:
      should_run: ${{ steps.should_run.outputs.should_run }}
    steps:
      - uses: actions/checkout@v2
      - name: print latest_commit
        run: echo ${{ github.sha }}

      - id: should_run
        continue-on-error: true
        name: check latest commit is less than a day
        if: ${{ github.event_name == 'schedule' }}
        run: test -z $(git rev-list  --after="24 hours"  ${{ github.sha }}) &amp;&amp; echo "::set-output name=should_run::false"
</code></pre>
<ol start="2">
<li>add this to all other jobs in your workflow:</li>
</ol>
<pre><code class="lang-auto">    needs: check_date
    if: ${{ needs.check_date.outputs.should_run != 'false' }}
</code></pre>
<p>for example:</p>
<pre><code class="lang-auto">  do_something:
    needs: check_date
    if: ${{ needs.check_date.outputs.should_run != 'false' }}
    runs-on: windows-latest
    name: do something.
    steps:
      - uses: actions/checkout@v2
</code></pre>

**Reply by OndrejSpanel** [reply]
This seems nice and simple, but when you push two commits quickly after each other, and then nothing for a week, no workflow will be executed for a week, until someone commits something.
